### ISSUE TITLE
index out of bounds: the len is 0 but the index is 0 at analyzer/src/traversal/expressions.rs:1300:26

### ISSUE THREAD

**Issue by agroce** [ethereum/fe#447]
Compiling this file:

```
contract Foo
 pub def s:
  o:Foo=Foo.create()
```

```
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', analyzer/src/traversal/expressions.rs:1300:26
stack backtrace:
   0:     0x56526fae2a00 - std::backtrace_rs::backtrace::libunwind::trace::h63b7a90188ab5fb3
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x56526fae2a00 - std::backtrace_rs::backtrace::trace_unsynchronized::h80aefbf9b851eca7
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x56526fae2a00 - std::sys_common::backtrace::_print_fmt::hbef05ae4237a4d72
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x56526fae2a00 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h28abce2fdb9884c2
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x56526fafbd3f - core::fmt::write::h3b84512577ca38a8
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/fmt/mod.rs:1092:17
   5:     0x56526fae0492 - std::io::Write::write_fmt::h465f8feea02e2aa1
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/io/mod.rs:1572:15
   6:     0x56526fae4b95 - std::sys_common::backtrace::_print::h525280ee0d29bdde
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x56526fae4b95 - std::sys_common::backtrace::print::h1f0f5b9f3ef8fb78
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x56526fae4b95 - std::panicking::default_hook::{{closure}}::ha5838f6faa4a5a8f
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:208:50
   9:     0x56526fae4643 - std::panicking::default_hook::hfb9fe98acb0dcb3b
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:225:9
  10:     0x56526ef7e3c3 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h858630bc94628395
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/alloc/src/boxed.rs:1560:9
  11:     0x56526f001160 - fe_compiler::errors::report_ice::h0561b22e4a0793bb
                               at /home/user/fe/compiler/src/errors.rs:74:5
  12:     0x56526f000e7e - fe_compiler::errors::DEFAULT_PANIC_HOOK::{{closure}}::{{closure}}::hbebb7f70ffef6630
                               at /home/user/fe/compiler/src/errors.rs:13:41
  13:     0x56526fae51c0 - std::panicking::rust_panic_with_hook::hb89f5f19036e6af8
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:595:17
  14:     0x56526fae4d37 - std::panicking::begin_panic_handler::{{closure}}::h119e7951427f41da
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:497:13
  15:     0x56526fae2ebc - std::sys_common::backtrace::__rust_end_short_backtrace::hce386c44bf47a128
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:141:18
  16:     0x56526fae4c99 - rust_begin_unwind
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:493:5
  17:     0x56526ef0e381 - core::panicking::panic_fmt::h2242888e8769cd33
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/panicking.rs:92:14
  18:     0x56526ef0e342 - core::panicking::panic_bounds_check::hd3a5ad32cddc7d96
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/panicking.rs:69:5
  19:     0x56526f885abe - <usize as core::slice::index::SliceIndex<[T]>>::index::h59dd17f5214e3d5c
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/slice/index.rs:182:10
  20:     0x56526f89c0e7 - core::slice::index::<impl core::ops::index::Index<I> for [T]>::index::hb6e0ed987955d227
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/slice/index.rs:15:9
  21:     0x56526f834dd1 - <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index::h8984d54746a7219a
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/alloc/src/vec/mod.rs:2381:9
  22:     0x56526f8468a4 - fe_analyzer::traversal::expressions::expr_call_type_attribute::h69695944e9d4dded
                               at /home/user/fe/analyzer/src/traversal/expressions.rs:1300:26

You've hit an internal compiler error. This is a bug in the Fe compiler.
Fe is still under heavy development, and isn't yet ready for production use.

If you would, please report this bug at the following URL:
  https://github.com/ethereum/fe/issues/new
Aborted
```

Using

```
[profile.dev]
debug = true
panic = "abort"
```

for fuzzing.  Using https://github.com/agroce/afl-compiler-fuzzer

**Comment by agroce** [comment]
Here's what I think is a variant:

```
contract Foo
 pub def s:
  o:Foo=Foo.create2(1)
```

```
thread 'main' panicked at 'index out of bounds: the len is 1 but the index is 1', analyzer/src/traversal/expressions.rs:1271:43
stack backtrace:
   0:     0x556f35e1ba00 - std::backtrace_rs::backtrace::libunwind::trace::h63b7a90188ab5fb3
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x556f35e1ba00 - std::backtrace_rs::backtrace::trace_unsynchronized::h80aefbf9b851eca7
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x556f35e1ba00 - std::sys_common::backtrace::_print_fmt::hbef05ae4237a4d72
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x556f35e1ba00 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h28abce2fdb9884c2
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x556f35e34d3f - core::fmt::write::h3b84512577ca38a8
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/fmt/mod.rs:1092:17
   5:     0x556f35e19492 - std::io::Write::write_fmt::h465f8feea02e2aa1
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/io/mod.rs:1572:15
   6:     0x556f35e1db95 - std::sys_common::backtrace::_print::h525280ee0d29bdde
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x556f35e1db95 - std::sys_common::backtrace::print::h1f0f5b9f3ef8fb78
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x556f35e1db95 - std::panicking::default_hook::{{closure}}::ha5838f6faa4a5a8f
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:208:50
   9:     0x556f35e1d643 - std::panicking::default_hook::hfb9fe98acb0dcb3b
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:225:9
  10:     0x556f352b73c3 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h858630bc94628395
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/alloc/src/boxed.rs:1560:9
  11:     0x556f3533a160 - fe_compiler::errors::report_ice::h0561b22e4a0793bb
                               at /home/user/fe/compiler/src/errors.rs:74:5
  12:     0x556f35339e7e - fe_compiler::errors::DEFAULT_PANIC_HOOK::{{closure}}::{{closure}}::hbebb7f70ffef6630
                               at /home/user/fe/compiler/src/errors.rs:13:41
  13:     0x556f35e1e1c0 - std::panicking::rust_panic_with_hook::hb89f5f19036e6af8
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:595:17
  14:     0x556f35e1dd37 - std::panicking::begin_panic_handler::{{closure}}::h119e7951427f41da
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:497:13
  15:     0x556f35e1bebc - std::sys_common::backtrace::__rust_end_short_backtrace::hce386c44bf47a128
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/sys_common/backtrace.rs:141:18
  16:     0x556f35e1dc99 - rust_begin_unwind
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/std/src/panicking.rs:493:5
  17:     0x556f35247381 - core::panicking::panic_fmt::h2242888e8769cd33
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/panicking.rs:92:14
  18:     0x556f35247342 - core::panicking::panic_bounds_check::hd3a5ad32cddc7d96
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/panicking.rs:69:5
  19:     0x556f35bbeabe - <usize as core::slice::index::SliceIndex<[T]>>::index::h59dd17f5214e3d5c
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/slice/index.rs:182:10
  20:     0x556f35bd50e7 - core::slice::index::<impl core::ops::index::Index<I> for [T]>::index::hb6e0ed987955d227
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/core/src/slice/index.rs:15:9
  21:     0x556f35b6ddd1 - <alloc::vec::Vec<T,A> as core::ops::index::Index<I>>::index::h8984d54746a7219a
                               at /rustc/9bc8c42bb2f19e745a63f3445f1ac248fb015e53/library/alloc/src/vec/mod.rs:2381:9
  22:     0x556f35b7f3f0 - fe_analyzer::traversal::expressions::expr_call_type_attribute::h69695944e9d4dded
                               at /home/user/fe/analyzer/src/traversal/expressions.rs:1271:43

You've hit an internal compiler error. This is a bug in the Fe compiler.
Fe is still under heavy development, and isn't yet ready for production use.

If you would, please report this bug at the following URL:
  https://github.com/ethereum/fe/issues/new
Aborted
```

**Comment by agroce** [comment]
Also:

```
contract Foo
 pub def s:
  o:Foo=Foo.create2()
```

**Comment by agroce** [comment]
Another variant:

```
contract Bar:
    pub fn foo():
        Bar.create2( 0>1)
```

**Comment by sbillig** [comment]
This will be fixed by a change in made in #577

**Comment by sbillig** [comment]
Fixed in #577
